### DISCUSSION TITLE
[Java] Is there an option for max heap size when running semmle-extractor-java.jar?

### DISCUSSION THREAD

**Discussion by brandonshiyay** [github/securitylab#416] [General] [answered]
Hi, I was trying to create a database for an internal project and failed due to an OutOfMemoryError. 
![screenshoot](https://user-images.githubusercontent.com/43565071/127818215-2c210875-a5b2-4022-9ee8-c9c2068ef6df.png)

Since it's a heap size problem, I tried again with the `-J -Xmx4096M` flag to increase the max heap size a bit, but didn't really work as the process still failed with the same message. Is there a way to provide a JVM option for the extractor command?

**Comment by smowton** [comment, marked as answer]
Does https://github.com/github/codeql/issues/5292#issuecomment-787483542 solve your problem?

**Reply by brandonshiyay** [reply]
Yes, problem solved, thank you.
